Library/Homebrew/os/linux/global.rb

```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
+HOMEBREW_READELF_RB = ENV["HOMEBREW_READELF_RB"]
+HOMEBREW_PATCHELF_RB = ENV["HOMEBREW_PATCHELF_RB"]
+require "patchelf" if HOMEBREW_PATCHELF_RB || HOMEBREW_READELF_RB
```

Comment:
```suggestion
if HOMEBREW_PATCHELF_RB || HOMEBREW_READELF_RB
  install_gem! "patchelf"
  require "patchelf"
end
```

Reply:
i'm facing a small problem,
`install_gem!` is not ready until `utils/gems` is required, which occurs after including this module in global module. 

It fails with `undefined method: install_gem!` for most commands.
few of the commands like `brew style` install the gem.

i'll figure out a workaround, if u have any quick suggestions that will be helpful. Thanks

Reply:
I'd suggest moving these constants away from `global.rb` and into e.g. `os/linux/elf.rb`. Another alternative would be just checking the environment variables and not set constants at all.